pyspecProcScripts/integrate_limits.py

```diff
@@ -1,43 +1,132 @@
 from pyspecdata import *
-from pylab import subplots
+from pylab import subplots, axvline
 import numpy as np
 from .fwhm_calculate import fwhm_calculator
 import logging
+from pylab import r_,fft,ifft,ifftshift,fftshift,exp,ones_like
+from matplotlib.pyplot import annotate
+logger = init_logging("debug")
 
+def integrate_limits(s, axis="t2",
+        filter_width=100,
+        Gaussian=False,
+        Lorentzian=True,
+        Lorentzian_to_Gaussian=False,
+        fl=None):
+    r"""
+    Integrate Limits
+    ============================
 
-def integrate_limits(s, axis="t2", fl=None):
-    """automatically determine the integration limits
+    This function takes data in the time domain and applies a matched filter
+    (choice of Lorentzian or Gaussian, default is Lorentzian) to the frequency
+    domain data, which it then uses to determine the integration limits based on a
+    cut off from the maximum signal intensity.
 
-    Parameters
-    ==========
-    axis: str
-        name of the dimension along which you want to determine the integration
-        limits
+    axisname: str
+        apply convolution along `axisname`
+
+    fwhm: int
+        width of the matched filter is `fwhm` - this is calculated more precisely
+        within the program
+
+    Gaussian: boolean
+        set `Gaussian` to True in order to calculate the matched-filter via
+        Gaussian convolution of the frequency domain signal (i.e., apply a matched
+        Gaussian filter to peak in frequency domain).
+        If `Gaussian` is set to True, set `Lorentzian` to False.
+
+    Lorentzian: boolean
+        set `Lorentzian` to True in order to calculate the matched-filter via
+        Lorentzian convolution of the frequency domain signal (i.e., apply a
+        matched Lorentzian filter to peak in frequency domain).
+        If `Lorentzian` is set to True, set `Gaussian` to False.
+
+
+    Lorentzian_to_Gaussian: boolean
+        set `Lorentzian_to_Gaussian` to True in order to apply this
+        transformation to your data, in order to determine integral limits.
+        Must set `Lorentzian` to True and `Gaussian` to False.
```

Comment:
make a single keyword with three string options

Reply:
ok, I have done this